diagnostics/jaeger/grpc_tracer.go

```diff
@@ -0,0 +1,192 @@
+	jaegerc "github.com/uber/jaeger-client-go"
+	"github.com/uber/jaeger-client-go/config"
+	"mosn.io/api"
+	"mosn.io/mosn/pkg/log"
+	"mosn.io/mosn/pkg/trace"
+	"mosn.io/mosn/pkg/trace/jaeger"
+	"mosn.io/mosn/pkg/types"
+
+	ltrace "mosn.io/layotto/components/trace"
+	"mosn.io/layotto/diagnostics/grpc"
+	"mosn.io/layotto/diagnostics/protocol"
+)
+
+const (
+	serviceName              = "layotto"
+	strategy                 = "strategy"
+	agentHost                = "agent_host"
+	collectorEndpoint        = "collector_endpoint"
+	defaultServiceName       = "layotto"
+	defaultJaegerAgentHost   = "127.0.0.1:6831"
+	jaegerAgentHostKey       = "TRACE"
+	appIDKey                 = "APP_ID"
+	defaultCollectorEndpoint = "http://127.0.0.1:14268/api/traces"
+	defaultStrategy          = "collector"
+)
+
+type grpcJaegerTracer struct {
+	tracer opentracing.Tracer
+}
+
+type grpcJaegerSpan struct {
+	*ltrace.Span
+	ctx        context.Context
+	trace      *grpcJaegerTracer
+	jaegerSpan opentracing.Span
+	spanCtx    jaegerc.SpanContext
+}
+
+func init() {
+	trace.RegisterTracerBuilder(jaeger.DriverName, protocol.Layotto, NewGrpcJaegerTracer)
+}
+
+func NewGrpcJaegerTracer(traceCfg map[string]interface{}) (api.Tracer, error) {
+	// 1. construct the ReporterConfig, which is used to communicate with jaeger
+	var reporter *config.ReporterConfig
+
+	// Determining whether to start the agent
+	strategy := getStrategy(traceCfg)
+
+	if strategy == defaultStrategy {
+		reporter = &config.ReporterConfig{
+			LogSpans:            false,
+			BufferFlushInterval: 1 * time.Second,
+			CollectorEndpoint:   getCollectorEndpoint(traceCfg),
+		}
+	} else {
+		reporter = &config.ReporterConfig{
+			LogSpans:            false,
+			BufferFlushInterval: 1 * time.Second,
+			LocalAgentHostPort:  getAgentHost(traceCfg),
+		}
+	}
+	// 2. construct the Configuration
+	cfg := config.Configuration{
+		Disabled: false,
+		Sampler: &config.SamplerConfig{
+			Type:  "const",
+			Param: 1,
+		},
+		Reporter: reporter,
+	}
+
+	cfg.ServiceName = getServiceName(traceCfg)
+
+	// 3. use the Configuration to construct a new tracer
+	tracer, _, err := cfg.NewTracer()
+
+	log.DefaultLogger.Infof("[layotto] [jaeger] [tracer] jaeger agent host:%s, report service name:%s",
+		getAgentHost(traceCfg), getServiceName(traceCfg))
+
+	if err != nil {
+		log.DefaultLogger.Errorf("[jaeger] [tracer] [layotto] cannot initialize Jaeger Tracer")
+		return nil, err
+	}
+
+	// 4. adapt to the `api.Tracer`
+	return &grpcJaegerTracer{
+		tracer: tracer,
+	}, nil
+}
+
+func getAgentHost(traceCfg map[string]interface{}) string {
+	if agentHost, ok := traceCfg[agentHost]; ok {
+		return agentHost.(string)
+	}
+
+	//if TRACE is not set, get it from the env variable
+	if host := os.Getenv(jaegerAgentHostKey); host != "" {
+		return host
+	}
+
+	return defaultJaegerAgentHost
+}
+
+func getStrategy(traceCfg map[string]interface{}) string {
+	if strategy, ok := traceCfg[strategy]; ok {
+		return strategy.(string)
```

Comment:
How about we check the strategy string here to make sure that it's `collector` or `agent`.
If it's a typo, e.g. `collect`, the `NewGrpcJaegerTracer` func can return an error